### PR TITLE
feat: deposit trade helper wrap and deposit

### PR DESCRIPTION
**Pull request by chrisling-dev**

### CODE REVIEW

**Review by chrisling-dev — COMMENTED** [review]
self reviewed

File: apps/frontend/core-dapp/src/stores/DepositTradeHelperStore.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    value: BigNumber
+  ): Promise<{ success: boolean; error?: string }> {
+    try {
+      const tx = await this.sendTransaction<WrapAndDeposit>('wrapAndDeposit', [recipient], {
+        value,
+      })
+      await tx.wait()
+      return { success: true }
+    } catch (e) {
+      return { success: false, error: (e as Error).message }
```
Comment (MeLlamoPablo) [inline apps/frontend/core-dapp/src/stores/DepositTradeHelperStore.ts:25]: Why don't we use `makeError` like in other places?

File: apps/frontend/core-dapp/src/stores/DepositTradeHelperStore.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+export class DepositTradeHelperStore extends ContractStore<RootStore, SupportedContracts> {
+  constructor(root: RootStore) {
+    super(root, 'DEPOSIT_TRADE_HELPER', DepositTradeHelperAbi__factory)
+  }
+
+  async wrapAndDeposit(
+    recipient: string,
+    value: BigNumber
+  ): Promise<{ success: boolean; error?: string }> {
```
Comment (MeLlamoPablo) [inline apps/frontend/core-dapp/src/stores/DepositTradeHelperStore.ts:17]:
```suggestion
  ): Promise<{ success: true; } | { success: false; error: string }> {
```

I'm a bit picky with these kind of things 😁 but feel free to disregard.

**Review by MeLlamoPablo — APPROVED** [review]
LGTM after nits

**Pull request merged**